### DISCUSSION TITLE
Broad Binary Rewriter Evaluation Results and Presentation

### DISCUSSION THREAD

**Discussion by eschulte** [microsoft/llvm-mctoll#178] [General]
I'm writing to share the results of a broad analysis of binary
rewriters, including mctoll (see the paper for details).  Note that this
is an ongoing evaluation that should be refreshed every year or so (see
the repo for details).  This work also poses a number of questions for
the binary rewriting community at large (see the end of the slides/talk
for specifics).

| what       | where                                                      |
|------------|------------------------------------------------------------|
| paper      | https://arxiv.org/abs/2203.13231                           |
| talk (13m) | https://cset22.isi.edu/videos/session4/binaryrewriters.mp4 |
| slides     | https://cset22.isi.edu/slides/binaryrewriters.pdf          |
| repo       | https://gitlab.com/GrammaTech/lifter-eval                  |
| artifacts  | https://gitlab.com/GrammaTech/lifter-eval-artifacts        |

I'd love to hear the mctoll developer's thoughts on everything from the evaluation, to
the questions and suggestions in the talk/slides.

Thanks!

**Comment by bharadwajy** [comment]
@eschulte  Thank you very much sharing the paper and related details. Also thank you for including `llvm-mctoll` in the evaluation. I look forward to its continued inclusion in the ongoing evaluation as we make progress in improving the tool's functionality.

I plan to read through the material in-depth and share feedback, if I have any. Equally importantly, I look forward to reading about the  learnings from other efforts as well as become aware of the questions currently of relevance to the binary rewriting community at large.
